examples/demo-rollup/src/main.rs

```diff
@@ -139,23 +119,22 @@ async fn main() -> Result<(), anyhow::Error> {
         ledger_rpc::get_ledger_rpc::<DemoBatchReceipt, DemoTxReceipt>(ledger_db.clone());
     methods
         .merge(ledger_rpc_module)
-        .expect("Failed to merge rpc modules");
+        .expect("Failed to merge ledger RPC modules");
+
+    let batch_builder = demo_runner.take_batch_builder()?;
```

Comment:
Looking at this API, it seems a little odd that the demo-runner is opinionated about only creating a single builder. Is there something that makes it impossible for the demo to create two builders? If not, maybe we should make this API infallible and rename to somethign `new_batch_builder()`

Reply:
I decided to go this way, as it solves problem of internal set of pending transactions, I call it mempool for simplicity.
So it can only takes builder once.

I see several ways how to do it:

1. method signature changes to `fn get_batch_builder(&self) -> Arc<BacthBuilder>`, but since it is shared, and requires mutable references to do any operation, that can be done with `Arc<Mutex<BatchBuilder>>`
2. internal mempool of BatchBuilder is wrapped in `Arc<Mutex<>`
3. We allow several independent BatchBuilder with their own mempools
4. Actual batch builder is started in separate thread, and this method returns a struct that just channels requests to actual mempool.

Each options has its own drawbacks